src/Microsoft.ApplicationInsights.AspNet/TelemetryInitializers/SyntheticTelemetryInitializer.cs

```diff
@@ -0,0 +1,60 @@
+﻿namespace Microsoft.ApplicationInsights.AspNet.TelemetryInitializers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Net.Sockets;
+    using Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.AspNet.Http;
+    using Microsoft.AspNet.Http.Features;
+
+    /// <summary>
+    /// This will allow to mark synthetic traffic from availability tests
+    /// </summary>
+    public class SyntheticTelemetryInitializer : TelemetryInitializerBase
+    {
+        private const string HeaderNameDefault = "GsmSyntheticTestRunId";
```

Comment:
The issue #23 specified the need to check for that specific header. If it is present or not I believe it would depend on the originator of the request to be tracked. 

Is there a different header that needs to be checked?